### ISSUE TITLE
unable to import GPG keys if bit 7 "critical" of the subpacket type is set

### ISSUE THREAD

**Issue by bb-Ricardo** [rpm-software-management/rpm#2323]
Hi,

I try to debug an issue we ran into while trying to import GPG keys generate using GO https://github.com/ProtonMail/go-crypto

The sub package type "Issuer ID" is always set "critical": https://github.com/ProtonMail/go-crypto/blob/cf6655e29de41fd667b667c8bc291f7dac05a393/openpgp/packet/signature.go#L845

According to RFC https://www.rfc-editor.org/rfc/rfc4880#section-5.2.3.1 the evaluation should fail if the critical bit is set **AND** the evaluation for this sub packet type is **NOT** implemented.

And this line should take care of this: https://github.com/rpm-software-management/rpm/blob/8e6108a5964c7289f3db70f3d188293276416528/rpmio/rpmpgp_internal.c#L369-L370

Unfortunately with the latest version of SuSE Linux this validation somehow triggers wrong results.

I'm not sure where the culprit lies here.
Sample keys to test it can be found here: https://github.com/ProtonMail/go-crypto/issues/134
The rpm version used here is `RPM version 4.14.3`

 Thank you

**Comment by DemiMarie** [comment]
Please report this to SUSE.  RPM 4.14.3 won’t be getting updates from upstream.

**Comment by bb-Ricardo** [comment]
Thank you again. Found that the issue was introduced in SuSE Linux between 15.1 and 15.2

**Comment by mlschroe** [comment]
(I think this should already be fixed with the latest maintenance update. I'm not sure if it is already released, though.)

**Comment by mlschroe** [comment]
`rpm -q --changelog rpm` needs to include:
```
- Strip critical bit in signature subpackage parsing
  * modified patch: pgpharden.diff
```

**Comment by mlschroe** [comment]
It should be available in 15.3 and 15.4.
